packages/mobile/src/components/buy-sell/TokenAmountSection.tsx

```diff
@@ -0,0 +1,309 @@
+        {formattedAvailableBalance}
+      </Text>
+    </Flex>
+  )
+}
+
+const StackedBalanceSection = ({
+  formattedAvailableBalance,
+  tokenInfo,
+  isStablecoin
+}: BalanceSectionProps) => {
+  const { symbol } = tokenInfo
+  const { cornerRadius } = useTheme()
+  const TokenIcon = symbol === 'AUDIO' ? IconTokenAUDIO : IconLogoCircleUSDC
+
+  if (!formattedAvailableBalance || !TokenIcon) {
+    return null
+  }
+
+  return (
+    <Flex direction='row' alignItems='center' gap='s'>
+      <Flex direction='column' alignItems='flex-end'>
+        <Text variant='heading' size='s' color='subdued'>
+          {messages.tokenTicker(symbol, !!isStablecoin)}
+        </Text>
+        <Text variant='title' size='s' color='default'>
+          {messages.stackedBalance(formattedAvailableBalance)}
+        </Text>
+      </Flex>
+      <TokenIcon size='4xl' style={{ borderRadius: cornerRadius.circle }} />
+    </Flex>
+  )
+}
+
+const CryptoAmountSection = ({
+  formattedAmount,
+  tokenInfo,
+  isStablecoin,
+  priceDisplay
+}: {
+  formattedAmount: string
+  tokenInfo: TokenInfo
+  isStablecoin: boolean
+  priceDisplay?: string
+}) => {
+  const { spacing, cornerRadius } = useTheme()
+  const { symbol } = tokenInfo
+  const TokenIcon = symbol === 'AUDIO' ? IconTokenAUDIO : IconLogoCircleUSDC
+  const tokenTicker = messages.tokenTicker(symbol, !!isStablecoin)
+
+  if (!TokenIcon) {
+    return null
+  }
+
+  return (
+    <Flex direction='row' alignItems='center' gap='s'>
+      <TokenIcon
+        style={{
+          borderRadius: cornerRadius.circle,
+          width: spacing.unit16,
+          height: spacing.unit16
+        }}
+      />
+      <Flex direction='column'>
+        <Flex direction='row' alignItems='center' gap='xs'>
+          <Text variant='heading' size='l'>
+            {formattedAmount}
+          </Text>
+          <Text variant='heading' size='l' color='subdued'>
+            {tokenTicker}
+          </Text>
+        </Flex>
+        {priceDisplay && (
+          <Text variant='heading' size='s' color='subdued'>
+            {priceDisplay}
+          </Text>
+        )}
+      </Flex>
+    </Flex>
+  )
+}
+
+export const TokenAmountSection = ({
+  title,
+  tokenInfo,
+  isInput,
+  amount,
+  onAmountChange,
+  onMaxClick,
+  availableBalance,
+  exchangeRate,
+  placeholder = '0.00',
+  isDefault = true,
+  error,
+  errorMessage,
+  tokenPrice,
+  isTokenPriceLoading,
+  tokenPriceDecimalPlaces = 2
+}: TokenAmountSectionProps) => {
+  const { spacing } = useTheme()
+  const { symbol, isStablecoin } = tokenInfo
+  const { formattedAvailableBalance, formattedAmount } =
+    useTokenAmountFormatting({
+      amount,
+      availableBalance,
+      exchangeRate,
+      isStablecoin: !!isStablecoin,
+      placeholder
+    })
+
+  const priceDisplay =
+    tokenPrice && !isTokenPriceLoading
+      ? toUSD(messages.tokenPrice(tokenPrice, tokenPriceDecimalPlaces))
+      : undefined
+
+  const titleText = useMemo(() => {
```

Comment:
i know ai likes to useMemo a bunch of stuff, but i believe when it comes to jsx it's not much more peformant